src/frontend/screens/Library/components/SteamInstall/SteamInstallButton.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { Tooltip, TooltipProps } from '@mantine/core'
+import { faSteam } from '@fortawesome/free-brands-svg-icons'
+import { Alert, Button } from '@hyperplay/ui'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import ContextProvider from 'frontend/state/ContextProvider'
+import { useTranslation } from 'react-i18next'
+import { observer } from 'mobx-react-lite'
+import libraryState from 'frontend/state/libraryState'
+import { WineInstallation } from 'common/types'
+import SteamInstallDialog from './SteamInstallDialog'
+import { launch } from 'frontend/helpers'
+
+const tooltipProps: Partial<TooltipProps> = {
+  offset: 16,
+  position: 'bottom',
+  withArrow: true,
+  className: 'Tooltip menu',
+  arrowSize: 10,
+  multiline: true,
+  w: 320,
+  transitionProps: { transition: 'fade-down', duration: 150 }
+}
+
+export default observer(function SteamInstallButton() {
+  const { platform, showDialogModal } = useContext(ContextProvider)
+  const [isInstalling, setIsInstalling] = useState(false)
+  const [showInstallDialog, setShowInstallDialog] = useState(false)
+  const [isCompatibilityLayerAvailable, setIsCompatibilityLayerAvailable] =
+    useState(false)
+  const [showAlert, setShowAlert] = useState<'success' | 'danger' | 'none'>(
+    'none'
+  )
+
+  const isMac = platform === 'darwin'
+  const isSteamInstalled =
+    isMac &&
+    libraryState.sideloadedLibrary.find((game) => game.app_name === 'steam')
+
+  const { t } = useTranslation()
+  useEffect(() => {
+    const checkWine = async () => {
+      const winelist: WineInstallation[] = await window.api.getAlternativeWine()
+      if (winelist.length > 0) {
+        setIsCompatibilityLayerAvailable(true)
+      }
+    }
+    checkWine()
+  }, [])
```

Comment:
I think this is fine, but if you need to check the loading status of this you could do:

```tsx
const { data: winelist, isLoading, isError } = useQuery<WineInstallation[]>(
    'alternativeWine',
    async () => {
      return await window.api.getAlternativeWine();
    }
  );

  const isCompatibilityLayerAvailable = winelist && winelist.length > 0;
```

Reply:
I have some TS issues with this code so probably more changes than that will be necessary:
![Uploading image.png…]()